packages/core/src/engine/EngineUtils.ts

```diff
@@ -265,6 +216,136 @@ function mapPalette<T, S>(f: (arg: T) => S, v: IPaletteV<T>): IPaletteV<S> {
 
 // Utils for converting types of values
 
+export class PathBuilder {
```

Comment:
Feels a little weird to put this in utils. Do you think `renderer/ShapeDef` is better for this?

Reply:
It does seem a bit weird here. ShapeDef looks like it defines the interfaces for each shape that can be used in style. This class doesn't get used in Style programs (it is used in Functions.ts), so maybe another spot? Maybe it could even have its own file since we don't have any other "svg builders" implemented yet?

Reply:
`shapes/PathBuilder.ts`?